src/Illuminate/Pipeline/PipelineServiceProvider.php

```diff
@@ -16,7 +16,15 @@ class PipelineServiceProvider extends ServiceProvider implements DeferrableProvi
     public function register()
     {
         $this->app->singleton(
-            PipelineHubContract::class, Hub::class
+            PipelineHubContract::class,
+            Hub::class
+        );
+
+        $this->app->singleton(
```

Comment:
Why is this a singleton? Wouldn't it keep pipes from previous usage? `->through()` replace the pipes, but `->pipe()` doesn't

Reply:
Very true point!

Reply:
Currently the Facade still kind of acts as a singleton due to the resolved instances being cached on a static property as seen here:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/Facade.php#L228-L241

To work around this you would also need to change the static `$cached` property to `false`:
https://github.com/laravel/framework/blob/10.x/src/Illuminate/Support/Facades/Facade.php#L31-L36